### PR TITLE
fix(ontology): support subproperty in ontology editor (DEV-332)

### PR DESCRIPTION
**Pull request by kilchenmann**
resolves DEV-332

### CODE REVIEW

File: src/app/project/ontology/default-data/default-properties.ts [hunk 1]
```diff
@@ -112,7 +124,7 @@ export class DefaultProperties {
                 //     description: 'A period of time between two dates',
                 //     subPropOf: Constants.HasValue,
                 //     objectType: Constants.DateValue,
-                //     guiEle: Constants.SalsahGui + Constants.HashDelimiter + 'Date',
+                //     guiEle: Constants.GuiDatePicker',
```
Comment (mdelez) [inline src/app/project/ontology/default-data/default-properties.ts:127]: is this commented out code going to be needed in the future or can we remove it?

File: src/app/project/ontology/ontology.service.ts [hunk 2]
```diff
@@ -79,4 +95,92 @@ export class OntologyService {
...
+                    // get property iri from another ontology
+                    this._cache.get('currentProjectOntologies').subscribe(
+                        (ontologies: ReadOntology[]) => {
+                            const onto = ontologies.find(i => i.id === baseOntoIri);
+                            superPropIri = onto.properties[subProp].subPropertyOf[0];
+                        }
+                    );
+                }
+
+                if (superPropIri) {
```
Comment (mdelez) [inline src/app/project/ontology/ontology.service.ts:121]: Why does this if statement need to be here?

File: src/app/project/ontology/ontology.service.ts [hunk 3]
```diff
@@ -79,4 +95,92 @@ export class OntologyService {
...
+                            ));
+                        } else {
+                            propType = (group.elements.find(i =>
+                                i.guiEle === property.guiElement && i.objectType === property.objectType
+                            ));
+                        }
+
+                    }
+                }
+                if (propType) {
```
Comment (mdelez) [inline src/app/project/ontology/ontology.service.ts:170]: same as my comment above, maybe I'm missing something..

**Review by mdelez — CHANGES_REQUESTED** [review]
Everything works well for me! I just have a few remarks/questions

File: src/app/project/ontology/default-data/default-properties.ts (hunk 1 above)
Comment (kilchenmann) [reply src/app/project/ontology/default-data/default-properties.ts:127]: I thought there is or there will be a difference between a date and a date period. But in our case you can always create a date period from a "normal" date property. So, there's no additional date period property needed. I can remove this section.

File: src/app/project/ontology/ontology.service.ts (hunk 3 above)
Comment (kilchenmann) [reply src/app/project/ontology/ontology.service.ts:170]: As soon as the propType is defined (or found), we can exit the for-loop.

File: src/app/project/ontology/ontology.service.ts (hunk 2 above)
Comment (kilchenmann) [reply src/app/project/ontology/ontology.service.ts:121]: As soon as the superPropIri is defined (or found), we can exit the for-loop.

Comment (kilchenmann) [reply src/app/project/ontology/ontology.service.ts:121]: Maybe it's not needed. I think I found this solution somewhere on stackoverflow 🙈

Comment (mdelez) [reply src/app/project/ontology/ontology.service.ts:121]: I think it would work if you put the `break` on line 117, no?

Comment (kilchenmann) [reply src/app/project/ontology/ontology.service.ts:121]: You mean like this:
Screen Shot 2022-02-01 at 10 10 38

Then I got an error

Comment (mdelez) [reply src/app/project/ontology/ontology.service.ts:121]: Yeah that's what I meant, good to know it doesn't work like I thought it would 😅

**Review by mdelez — APPROVED** [review]

**Pull request merged**